src/index.js

```diff
@@ -208,93 +274,83 @@ class ExtractTextPlugin {
+                  for (const chunkModule of extractedChunk.getModules()) {
+                    extractedChunk.removeModule(chunkModule);
                   }
-                } else callback();
-              },
-              (err) => {
-                if (err) return callback(err);
-                callback();
-              }
-            );
-          },
-          (err) => {
-            if (err) return callback(err);
-            extractedChunks.forEach((extractedChunk) => {
-              if (isInitialOrHasNoParents(extractedChunk)) {
-                this.mergeNonInitialChunks(extractedChunk);
-              }
-            }, this);
-            extractedChunks.forEach((extractedChunk) => {
-              if (!isInitialOrHasNoParents(extractedChunk)) {
-                extractedChunk.forEachModule((module) => {
-                  extractedChunk.removeModule(module);
-                });
-              }
-            });
-            compilation.applyPlugins(
-              'optimize-extracted-chunks',
-              extractedChunks
-            );
-            callback();
-          }
-        );
-      });
-      compilation.plugin('additional-assets', (callback) => {
-        extractedChunks.forEach((extractedChunk) => {
-          if (extractedChunk.getNumberOfModules()) {
-            extractedChunk.sortModules((a, b) => {
-              if (!options.ignoreOrder && isInvalidOrder(a, b)) {
-                compilation.errors.push(
-                  new OrderUndefinedError(a.getOriginalModule())
-                );
-                compilation.errors.push(
-                  new OrderUndefinedError(b.getOriginalModule())
-                );
-              }
-              return getOrder(a, b);
-            });
-            const chunk = extractedChunk.originalChunk;
-            const source = this.renderExtractedChunk(extractedChunk);
-
-            const getPath = (format) =>
-              compilation
-                .getPath(format, {
-                  chunk,
-                })
-                .replace(
-                  /\[(?:(\w+):)?contenthash(?::([a-z]+\d*))?(?::(\d+))?\]/gi,
-                  function() {
-                    // eslint-disable-line func-names
-                    return loaderUtils.getHashDigest(
-                      source.source(),
-                      arguments[1],
-                      arguments[2],
-                      parseInt(arguments[3], 10)
-                    );
-                  }
-                );
+                }
+              });
+              compilation.hooks.optimizeExtractedChunks.call(extractedChunks);
+              callback();
+            }
+          );
+        }
+      );
+      compilation.hooks.additionalAssets.tapAsync(
+        'additional-assets',
+        (callback) => {
+          extractedChunks.forEach((extractedChunk) => {
+            if (extractedChunk.getNumberOfModules()) {
+              extractedChunk.sortModules((a, b) => {
```

Comment:
`sortModules` is deprecated aswell, not to important though (leave it if it anything else causes failures) but if you can refcator it to use a standard `Array.prototype.sort` instead it would be great 👍. Maybe `chunk.getModules().sort()` works here